### DISCUSSION TITLE
Different output file in version 7.0.9-24 vs. 7.0.11-13

### DISCUSSION THREAD

**Discussion by camillavoigt** [ImageMagick/ImageMagick#4758] [Help]
Input file: ![004477](https://user-images.githubusercontent.com/22515314/151187650-0c124e2c-ba81-4ad5-81c8-3c89fdfe5cda.jpg)

When using this command in version **7.0.9-24**, I get a perfect transparent output file:
`convert -colorspace sRGB -background transparent "C:\Users\cvk\Downloads\004477.jpg" -alpha set -channel a -clip -negate +clip +mask -negate -trim -channel rgba +repage +mask -colorspace rgb -set colorspace srgb -colorspace ycbcr -filter mitchell -define filter:lobes=0 -define filter:blur=0.8 -distort resize 960x960 -gravity center -extent 1024x1024 -colorspace srgb -set colorspace rgb -colorspace srgb -depth 8 "C:\Users\cvk\Downloads\004477_NEW.png"`

**Output file:**
![004477_NEW](https://user-images.githubusercontent.com/22515314/151188639-36358e9f-e966-4cc9-a0e3-89813d9a3b48.png)

**Red background added to show the transparency:** 
![image](https://user-images.githubusercontent.com/22515314/151188846-fdd174ef-9e04-4e84-ae20-24430c0b31a8.png)

But when using the exact same command in version **7.0.11-13**, the output is totally different.

**Output file:**
![004477_NEW_wrong](https://user-images.githubusercontent.com/22515314/151189228-aeda408f-1931-4419-a5b4-131e547a9b5d.png)

**Red background added to show the **missing** transparency:** 
![image](https://user-images.githubusercontent.com/22515314/151189364-bf0c00d1-c43d-493a-90b0-69369303c691.png)

Is this a bug in version 7.0.11-13?

**Comment by fmw42** [comment]
Can you post a simple command (as short as possible) that shows the difference?

**Reply by camillavoigt** [reply]
Hi,
Very sorry for the late response. I have been struggling with the transcode.

I have tried to shorten the transcode setting as much as possible:
`convert -colorspace sRGB -background transparent "C:\Users\cvk\Downloads\004477.jpg" -alpha set -channel a -clip -negate +clip +mask -negate -trim -channel rgba +repage +mask -colorspace rgb -set colorspace srgb -colorspace ycbcr -filter mitchell -define filter:lobes=0 -define filter:blur=0.8 -colorspace srgb -set colorspace rgb -colorspace srgb -depth "C:\Users\cvk\Downloads\CVK3.png"`

Output 7.0.9 vs. 7.0.11:
![Simple_transcode_compare](https://user-images.githubusercontent.com/22515314/154087228-65b252af-42b0-444a-974f-1dae4965cc08.png)

Do you know if it is possible to get the portable version of ImageMagick 7.0.9? I does not seem like it is available on ImageMagick.org.

**Comment by snibgo** [comment]
You don't show the input jpeg for that image, so we can't test your command.

Portable versions of IM on windows are at https://imagemagick.org/script/download.php#windows

**Reply by camillavoigt** [reply]
Sorry, I forgot to add it. This is the input file:
![004477](https://user-images.githubusercontent.com/22515314/154214927-4afd0398-2663-47ef-9f96-6f5ddb296da0.jpg)
 
The portable version of ImageMagick 7.0.9 is no longer available at https://imagemagick.org/script/download.php#windows

**Comment by fmw42** [comment]
In my opinion, you have way too many repeats of operators and settings and not in a a good order. Also you should be using magick not convert when using Imagemagick 7.  Also you -depth at the end and no value for it.  Here is a trimmed down command that should be what you want. However, it does not work to produce transparency.

Unix syntax:

```
magick lamp.jpg -colorspace sRGB -background transparent \
-alpha set -channel a -clip -negate +clip \
-negate -trim +repage -channel rgba \
-colorspace ycbcr -filter mitchell \
-define filter:lobes=0 -define filter:blur=0.8 \
-set colorspace ycbcr -colorspace srgb \
-depth 8 x.png

```

![x](https://user-images.githubusercontent.com/13458218/154307242-a71bef6d-6720-4d97-b0bc-5dcbd5b230b4.png)


So lets look at the basic command for getting the transparency from the clip path:

In IM 6, this works fine:

`convert lamp.jpg -alpha set -channel a -clip -negate +clip -negate +channel x6.png`

![x6](https://user-images.githubusercontent.com/13458218/154309521-01cd8c88-ed79-4759-b5f5-0ae2cc79c427.png)


However, it fails in IM 7:

`magick lamp.jpg -alpha set -channel a -clip -negate +clip -negate +channel x7.png`

![x7](https://user-images.githubusercontent.com/13458218/154309596-65ed467f-083f-49bd-8818-3cc0a242a76c.png)

**Comment by fmw42** [comment]
CORRECTION:

According to the Porting Page at https://imagemagick.org/script/porting.php#cli the clipping syntax for IM 7 is a bit different. The following works

`magick lamp.jpg -alpha transparent -clip -alpha opaque +clip x7.png`

![x7](https://user-images.githubusercontent.com/13458218/154322719-db77b346-a294-4c6f-9867-e930e57b0dd6.png)

So your syntax should be:

```
magick lamp.jpg -colorspace sRGB \
-alpha transparent -clip -alpha opaque +clip \
-trim +repage \
-colorspace ycbcr -filter mitchell \
-define filter:lobes=0 -define filter:blur=0.8 \
-set colorspace ycbcr -colorspace srgb \
-depth 8 x.png

```

![x](https://user-images.githubusercontent.com/13458218/154322796-51997436-ac2e-42b3-bc55-e98095313c3b.png)

**Comment by snibgo** [comment]
As fmw42 says. I will add a couple of points.

The lamp file is encoded in Adobe RGB. "-colorspace sRGB" will not convert it to sRGB, no matter how many times you do this. The only method to convert it is with "-profile", such as "-profile sRGB.icc".

To convert the input to sRGB with transparent background, I suggest one of two methods. I show Windows BAT syntax. Inkscape is on my system path.

Method 1:
```
magick ^
  lamp.jpg ^
  -alpha set -clip -alpha transparent +clip ^
  -profile sRGB.icc ^
  x1.png
```
The transparency is binary. Each pixel is either fully transparnt or fully opaque, which may not be satisfactory.

Method 2:
```
magick -quiet -format "%%[8BIM:1999,2998:#1]" lamp.jpg info:temppath.svg

magick ^
  lamp.jpg ^
  ( temppath.svg -channel RGB -negate +channel +write p.png  ) ^
  -alpha off -compose CopyOpacity -composite -trim +repage ^
  -profile sRGB.icc ^
  x2.png
```
temppath.svg is temporary, and can then be deleted. In bash, this method can be done with the first command embedded in the second.

The transparency is anti-aliased, which may be preferred to the first method.

**Comment by fmw42** [comment]
I assumed he added -colorspace sRGB in case he had some other image he wanted to process that was a CMYK jpg image

**Comment by snibgo** [comment]
Yes. If a CMYK image has an embedded profile, then again it should be converted to sRGB using "-profile".

**Comment by fmw42** [comment]
Actually he had -colorspace sRGB before reading the input. That would not be correct for the JPG, but if the input was CMYK PDF, then the -colorspace sRGB would be needed and proper.

**Comment by camillavoigt** [comment]
Hi everyone,

Thanks for all the help you provided. I have got it working in our Production Environment now, using Image Magick 7.1.0-25 and provided command lines.

Have a great weekend!
